Libraries/Opc.Ua.Gds.Client.Common/GlobalDiscoveryServerClient.cs

```diff
@@ -853,53 +853,56 @@ public TrustListDataType ReadTrustList(NodeId trustListId)
                 (byte)OpenFileMode.Read);
 
             uint fileHandle = (uint)outputArguments[0];
-            MemoryStream ostrm = new MemoryStream();
-
-            try
+            using (MemoryStream ostrm = new MemoryStream())
             {
-                while (true)
+
+                try
                 {
-                    int length = 4096;
+                    while (true)
+                    {
+                        int length = 4096;
 
-                    outputArguments = Session.Call(
-                        trustListId,
-                        Opc.Ua.MethodIds.FileType_Read,
-                        fileHandle,
-                        length);
+                        outputArguments = Session.Call(
+                            trustListId,
+                            Opc.Ua.MethodIds.FileType_Read,
+                            fileHandle,
+                            length);
 
-                    byte[] bytes = (byte[])outputArguments[0];
-                    ostrm.Write(bytes, 0, bytes.Length);
+                        byte[] bytes = (byte[])outputArguments[0];
+                        ostrm.Write(bytes, 0, bytes.Length);
 
-                    if (length != bytes.Length)
-                    {
-                        break;
+                        if (length != bytes.Length)
+                        {
+                            break;
+                        }
                     }
                 }
-            }
-            catch (Exception)
-            {
-                throw;
-            }
-            finally
-            {
-                if (IsConnected)
+                catch (Exception)
                 {
-                    Session.Call(
-                        trustListId,
-                        Opc.Ua.MethodIds.FileType_Close,
-                        fileHandle);
+                    throw;
+                }
+                finally
+                {
+                    if (IsConnected)
+                    {
+                        Session.Call(
+                            trustListId,
+                            Opc.Ua.MethodIds.FileType_Close,
+                            fileHandle);
+                    }
                 }
-            }
-
-            ostrm.Position = 0;
 
-            BinaryDecoder decoder = new BinaryDecoder(ostrm, Session.MessageContext);
-            TrustListDataType trustList = new TrustListDataType();
-            trustList.Decode(decoder);
-            decoder.Close();
-            ostrm.Close();
+                ostrm.Position = 0;
 
-            return trustList;
+                TrustListDataType trustList = new TrustListDataType();
+                using (BinaryDecoder decoder = new BinaryDecoder(ostrm, Session.MessageContext))
+                {
+                    trustList.Decode(decoder);
+                    decoder.Close();
+                }
+                ostrm.Close();
```

Comment:
ostrm.Close() and the decoder.Close() in 901 should be superfluous now, with the new using statement?

Reply:
it looks like decoder should also have a mandatory leaveopen flag for streams, but I'd rather address this in a future PR